xbmc/video/VideoDatabase.cpp

```diff
@@ -5875,11 +5873,9 @@ bool CVideoDatabase::GetSeasonsByWhere(const std::string& strBaseDir, const Filt
       int showId = m_pDS->fv(VIDEODB_ID_SEASON_TVSHOW_ID).get_asInt();
       int iSeason = m_pDS->fv(VIDEODB_ID_SEASON_NUMBER).get_asInt();
       std::string name = m_pDS->fv(VIDEODB_ID_SEASON_NAME).get_asString();
-      std::string path = m_pDS->fv(VIDEODB_ID_SEASON_TVSHOW_PATH).get_asString();
 
       if (mapSeasons.find(std::make_pair(showId, iSeason)) == mapSeasons.end() &&
-         (CProfilesManager::GetInstance().GetMasterProfile().getLockMode() == LOCK_MODE_EVERYONE || g_passwordManager.bMasterUser ||
-          g_passwordManager.IsDatabasePathUnlocked(path, *CMediaSourceSettings::GetInstance().GetSources("video"))))
+         (CProfilesManager::GetInstance().GetMasterProfile().getLockMode() == LOCK_MODE_EVERYONE || g_passwordManager.bMasterUser))
```

Comment:
I'd remove the locking stuff anyway TBH. After all, it only works if the first show path (or whatever one is returned) is locked anyway.

Also, if the show path is locked, you have to go out of your way to make it to the season list for that show without triggering the lock anyway, right (e.g. by linking direct to that folder, doing it via the webinterface or similar)? And if you can do that, I reckon you can get direct to the episode listing in the same way bypassing the lock.